### PR TITLE
PUD-220 swagger to typescript

### PR DESCRIPTION
**Pull request by jonnywyatt**
Assessed a few libraries and settled on [openapi-typescript-codegen](https://www.npmjs.com/package/openapi-typescript-codegen). It's capable of outputting what looks like a full Rest client with routes etc; worth a further look at that later.

To satisfy TS requirements, have manually created a blank index.d.ts in the models folder and renamed the index.ts file in @types/

### CODE REVIEW

File: server/clients/manageRecallsApi/manageRecallsApiClient.ts [hunk 1]
```diff
@@ -8,11 +9,6 @@ export interface PrisonerSearchResult {
   dateOfBirth?: string
 }
 
-export interface Recall {
-  id: string
-  nomsNumber: string
-}
-
 export interface Pdf {
```
Comment (AndyNewton) [inline server/clients/manageRecallsApi/manageRecallsApiClient.ts:12]: Can this be superseded by the `Pdf.ts` type?

**Review by AndyNewton — COMMENTED** [review]
Not sure whether a few lines in the README to explain how this works is worthwhile?
Can we foresee scenarios where this causes issues?  Thanks!

File: server/clients/manageRecallsApi/manageRecallsApiClient.ts (hunk 1 above)
Comment (jonnywyatt) [reply server/clients/manageRecallsApi/manageRecallsApiClient.ts]: Good shouts, done
I'd left the Pdf type as thought I'd apply the types in a later PR. The Recall one was just to prove it works. But yep simple to do now

**Review by AndyNewton — APPROVED** [review]
LGTM!

**Pull request merged**